### PR TITLE
Chapter 6: Modelling Users

### PR DESCRIPTION
**Pull request by rashidfirhat98**

### CODE REVIEW

**Comment by rashidfirhat98** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  uniqueness: true
+
+  validates :password, presence: true,
+  length: {minimum: Settings.user.password.min_length}
+
+  has_secure_password
+
+  before_save :downcase_email
+  private
+  def downcase_email
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  before_save :downcase_email
  
  private
  
  def downcase_email
```

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+  validates :name, presence: true,
+  length: {maximum: Settings.user.name.max_length}
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
    length: {maximum: Settings.user.name.max_length}
```

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  validates :email, presence: true,
+  length: {maximum: Settings.user.email.max_length},
+  format: {with: Settings.user.email.regex},
+  uniqueness: true
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
    length: {maximum: Settings.user.email.max_length},
    format: {with: Settings.user.email.regex},
    uniqueness: true
```

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  validates :password, presence: true,
+  length: {minimum: Settings.user.password.min_length}
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
    length: {minimum: Settings.user.password.min_length}
```

Comment (rashidfirhat98) [reply app/models/user.rb]: done

**Comment by rashidfirhat98** [comment]
ready

File: app/models/user.rb (hunk 3 above)
Comment (rashidfirhat98) [reply app/models/user.rb]: done

**Comment by manhnt-0960** [comment]
merged

**Pull request merged**